### PR TITLE
T33843 (Debian)

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
A pretty direct update. Only minor rebase conflicts.

https://phabricator.endlessm.com/T33843

### CODE REVIEW

**Comment by GeorgesStavracas** [comment]
Retest this please

**Comment by GeorgesStavracas** [comment]
retest this please

File: debian/control [hunk 1]
```diff
@@ -16,15 +16,15 @@ Build-Depends: debhelper-compat (= 13),
                gtk-doc-tools,
                libadwaita-1-dev (>= 1.0.1),
                libappstream-dev (>= 0.14.0),
-               libflatpak-dev (>= 1.14.0) [amd64 arm64 armel armhf i386 mipsel mips64el ppc64el s390x hppa powerpc ppc64 riscv64 x32],
+               libflatpak-dev [amd64 arm64 armel armhf i386 mipsel mips64el ppc64el s390x hppa powerpc ppc64 riscv64 x32],
```
Comment (wjt) [inline debian/control:19]: I was a little sceptical of this patch so I went looking for why Debian raised the dependency.

Here's the commit: https://salsa.debian.org/gnome-team/gnome-software/-/commit/562eb7ea73e362e561e154deefe5d5bf954519df

> ### Update dependency versions
> in preparation for the libsoup3 transition

Flatpak 1.14 introduced a curl backend…

File: debian/control [hunk 2]
```diff
@@ -16,15 +16,15 @@ Build-Depends: debhelper-compat (= 13),
...
                libfwupd-dev (>= 1.5.6) [linux-any],
                libglib2.0-dev (>= 2.70.0),
                libglib-testing-0-dev,
                libgnome-desktop-3-dev (>= 3.18.0),
                libgtk-4-dev (>= 4.6),
                libgudev-1.0-dev [linux-any],
                libjson-glib-dev (>= 1.6.0),
-               libostree-dev (>= 2022.5-2~) [linux-any],
+               libostree-dev [linux-any],
```
Comment (wjt) [inline debian/control:27]: and Debian's 2022.5-2 ostree package switched to the libcurl backend. OK!

**Review by wjt — APPROVED** [review]

File: debian/rules [hunk 3]
```diff
@@ -43,6 +43,19 @@ ifeq ($(DEB_HOST_ARCH_OS), linux)
 	endif
 endif
 
+# EOS-specific configuration
+GS_CONFIGURE_FLAGS += \
+	-Deos_updater=true \
+	-Dexternal_appstream=true \
+	-Dfwupd=false \
+	-Dgudev=false \
+	-Dhardcoded_popular=false \
```
Comment (wjt) [inline debian/rules:52]: This will fail to build.

See 0983f7392c204ed1ca6dba69061761401d1796aa.

File: debian/rules [hunk 4]
```diff
@@ -43,6 +43,19 @@ ifeq ($(DEB_HOST_ARCH_OS), linux)
...
+	-Dmalcontent=true \
+	-Dmogwai=true \
+	-Dpackagekit=false \
+	-Dsoup2=true \
+	-Dtests=true
```
Comment (wjt) [inline debian/rules:57]: and this will override d922fd66794c4b318c2965db62f0942ea77f2a9b on OBS and presumably again fail the build.

**Review by wjt — CHANGES_REQUESTED** [review]

File: debian/control (hunk 1 above)
Comment (GeorgesStavracas) [reply debian/control:19]: Correct. I've added this to the commit message. We will be able to drop this patch later.

File: debian/rules (hunk 4 above)
Comment (GeorgesStavracas) [reply debian/rules]: In theory tests are disabled a few lines below:

```
# DISABLED
override_dh_auto_test:
```

But removed nonetheless

**Review by wjt — APPROVED** [review]

**Comment by wjt** [comment]
Pushed a patch to fix:

```
dpkg-gencontrol: warning:     debian/changelog(l5): ignoring invalid week day 'Qui'
```

**Comment by wjt** [comment]
The resulting package builds but fails to install:

```
The following packages have unmet dependencies:
 gnome-software-plugin-flatpak : Depends: flatpak (>= 1.14.0) but 1.12.7+dev7.438e811-0bem1 is to be installed
E: Unable to correct problems, you have held broken packages.
```

**Pull request merged**